### PR TITLE
This fixes the problem with mixed variables in queries

### PR DESCRIPTION
**Pull request by annando**
This fixes the problem introduced here: https://github.com/friendica/friendica/pull/5267

That PR isn't the problem, but now we see the consequences and have to fix them as well.

### CODE REVIEW

**Comment by MrPetovan** [comment]
Won't this convert `[10, '10.1', 10.2]` to `[10, 10, 10]`?

**Comment by annando** [comment]
@MrPetovan f*** you are right. I have a look at it. We don't have floats at the moment, but the class should be save for this situation as well.

**Comment by MrPetovan** [comment]
I really wish we had unit testing for DB related stuff...

**Comment by MrPetovan** [comment]
Many unit test are failing with the following message:
`PDOException: SQLSTATE[HY000]: General error: 2014 Cannot execute queries while other unbuffered queries are active.  Consider using PDOStatement::fetchAll().  Alternatively, if your code is only ever going to run against mysql, you may enable query buffering by setting the PDO::MYSQL_ATTR_USE_BUFFERED_QUERY attribute.`

```
/home/travis/build/friendica/friendica/include/dba.php:412
/home/travis/build/friendica/friendica/include/dba.php:586
/home/travis/build/friendica/friendica/include/dba.php:878
/home/travis/build/friendica/friendica/src/Model/Item.php:1234
/home/travis/build/friendica/friendica/mod/item.php:733
/home/travis/build/friendica/friendica/include/api.php:1050
/home/travis/build/friendica/friendica/tests/ApiTest.php:1081
```

File: include/dba.php [hunk 1]
```diff
@@ -1296,6 +1299,39 @@ public static function buildCondition(array &$condition = [])
...
+								$is_int = true;
+							}
+
+							// To prevent to round floats we look for them
+							if (is_float($single_value)) {
+								$is_float = true;
+							}
+
+							// Is any non numeric value present?
+							if (!is_numeric($single_value)) {
```
Comment (MrPetovan) [inline include/dba.php]: `is_float('10.1') = false`
`is_numeric('10.1') = true`

**Pull request merged**